src/checked_transaction.rs

```diff
@@ -111,163 +122,13 @@ impl CheckedTransaction {
         self.checked_signatures
     }
 
-    pub const fn metadata(&self) -> Option<&Metadata> {
-        self.transaction.metadata()
-    }
-
-    pub fn tx_bytes(&mut self) -> Vec<u8> {
```

Comment:
I moved all methods below to `fuel-vm` directly because they are very specific and used only there. Plus they modify the inner `Transaction`, and it looks strange in the context of `CheckedTransaction`. `CheckedTransaction` means that it is checked and it is valid. `mut` methods can break the inner validity.

Reply:
Will this affect the `to_bytes()` method used here? https://github.com/FuelLabs/fuel-core/blob/master/fuel-core/src/executor.rs#L237

Reply:
The output will be the same but in different way=)